### ISSUE TITLE
Murmurhash64 vs wyhash

### ISSUE THREAD

**Issue by dumblob** [elucent/clover#1]
Just saw your tweet about murmurhash64 being faster than wyhash and yes, that is a known weakness of wyhash - namely batch speed (for loop with just the hash inside) is lower than a few other hashes.

OTOH e.g. hash tables do not need batch speed and rely on individual hash computations with non-negligible surrounding code. And there murmurhash (and all other "batch-optimized" hashes) is definitely slower. And of course both hashes were tested and optimized using traditional compilers, so if you use your own JIT, then I expect the results be different.

The bottom line is - hashes are unpredictable beasts and one needs to be very cautious (that is the reason why [smhasher](https://github.com/rurban/smhasher/ ) was made in the first place :wink:). I myself would only choose a hash after benchmarking it end-to-end (e.g. from clover and not from C/C++ backend code).

Feel free to close this after reading :wink:.
